src/main/java/uk/nhs/digital/nhsconnect/nhais/service/AbstractToEdifactService.java

```diff
@@ -0,0 +1,141 @@
+                .setTransactionTimestamp(translationItems.getTranslationTimestamp())
+                .setOperationId(translationItems.getOperationId());
+        outboundStateRepository.save(outboundState);
+    }
+
+    protected void addSequenceNumbersToSegments(T translationItems) {
+        for (Segment segment : translationItems.getSegments()) {
+            if (segment instanceof InterchangeHeader) {
+                InterchangeHeader interchangeHeader = (InterchangeHeader) segment;
+                interchangeHeader.setSequenceNumber(translationItems.getSendInterchangeSequence());
+            } else if (segment instanceof InterchangeTrailer) {
+                InterchangeTrailer interchangeTrailer = (InterchangeTrailer) segment;
+                interchangeTrailer.setSequenceNumber(translationItems.getSendInterchangeSequence());
+            } else if (segment instanceof MessageHeader) {
+                MessageHeader messageHeader = (MessageHeader) segment;
+                messageHeader.setSequenceNumber(translationItems.getSendMessageSequence());
+            } else if (segment instanceof MessageTrailer) {
+                MessageTrailer messageTrailer = (MessageTrailer) segment;
+                messageTrailer.setSequenceNumber(translationItems.getSendMessageSequence());
+            } else if (segment instanceof ReferenceTransactionNumber) {
+                ReferenceTransactionNumber referenceTransactionNumber = (ReferenceTransactionNumber) segment;
+                referenceTransactionNumber.setTransactionNumber(translationItems.getTransactionNumber());
+            } else if (segment instanceof DateTimePeriod) {
+                DateTimePeriod dateTimePeriod = (DateTimePeriod) segment;
+                if (dateTimePeriod.getTypeAndFormat().equals(DateTimePeriod.TypeAndFormat.TRANSLATION_TIMESTAMP)) {
+                    dateTimePeriod.setTimestamp(translationItems.getTranslationTimestamp());
+                }
+            }
+        }
+    }
+
+    protected OutboundMeshMessage translateInterchange(T translationItems) throws EdifactValidationException {
+        List<String> segmentStrings = new ArrayList<>(translationItems.getSegments().size());
+        for (Segment segment : translationItems.getSegments()) {
+            segmentStrings.add(segment.toEdifact());
+        }
+        String edifact = String.join("\n", segmentStrings);
+        MeshMessage meshMessage = new MeshMessage();
+        meshMessage.setContent(edifact);
+        meshMessage.setWorkflowId(WorkflowId.REGISTRATION);
+        meshMessage.setOperationId(translationItems.getOperationId());
+        meshMessage.setHaTradingPartnerCode(translationItems.getRecipient());
+        return meshMessage;
+    }
+
+    protected String getRecipientTradingPartnerCode(String recipient) throws FhirValidationException {
+        if (recipient.length() == 2) {
+            return recipient + "01";
+        } else {
+            return recipient + "1";
+        }
+    }
```

Comment:
Can recipient trading partner code have length of 4 characters? I've run into a problem when `TES5` was the recipient